cni/scripts/validate_state.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+function find_in_array() {
+    for i in $1
+    do
+        if [ "$i" == "$2" ] ; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+for node in $(kubectl get nodes -o name);
+do
+    echo "Current : $node"
+    node_name="${node##*/}"
+    node_ip=$(kubectl get "$node"  -o jsonpath='{$.status.addresses[?(@.type=="InternalIP")].address}')
+    echo "Node internal ip: $node_ip"
+    privileged_pod=$(kubectl get pods -n kube-system -o wide | grep privileged | grep $node_name | awk '{print $1}')
+    echo "privileged pod : $privileged_pod"
+    until cat azure_endpoints.json
+    do
+        kubectl exec -i $privileged_pod -n kube-system -- bash -c "cat /var/run/azure-cns/azure-endpoints.json" > azure_endpoints.json
+        sleep 10
+    done
+
+    cilium_agent=$(kubectl get pod -l k8s-app=cilium -n kube-system -o wide | grep $node_name | awk '{print $1}')
+    echo "cilium agent : $cilium_agent"
+    
+    until cat cilium_endpoints.json
+    do
+        kubectl exec -i $cilium_agent -n kube-system -- bash -c "cilium endpoint list -o json" > cilium_endpoints.json
+        sleep 10
+    done
+
+    total_pods=$(kubectl get pods --all-namespaces -o wide --field-selector spec.nodeName=$node_name,status.phase=Running --output json)
+
+    echo "Checking if there are any pods with no ips"
+    pods_with_no_ip=$(echo $total_pods | jq -j '(.items[] | select(.status.podIP == "" or .status.podIP == null))')
+    if [ "$pods_with_no_ip" != "" ]; then
+        echo "There are some pods with no ip assigned."
+        kubectl get pods -A
```

Comment:
maybe `-o wide` here too so we can see the node + IP

Reply:
Okay